sdk/core/Azure.Core/src/Operation.cs

```diff
@@ -81,6 +82,33 @@ public abstract class Operation
         /// </remarks>
         public abstract ValueTask<Response> WaitForCompletionResponseAsync(TimeSpan pollingInterval, CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Periodically calls the server till the long-running operation completes.
+        /// </summary>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> used for the periodical service calls.</param>
+        /// <returns>The last HTTP response received from the server.</returns>
+        /// <remarks>
+        /// This method will periodically call UpdateStatusAsync till HasCompleted is true, then return the final response of the operation.
+        /// </remarks>
+        public virtual Response WaitForCompletionResponse(CancellationToken cancellationToken = default)
+            => this.DefaultWaitForCompletionResponse(cancellationToken);
```

Comment:
I could be missing something, but why are the async `WaitForCompletion` methods abstract, while we provide default implementations to the sync ones?

Reply:
To avoid a breaking change.

Reply:
I could be wrong, but I think that changing a method from abstract to virtual is not a breaking change. If I'm not missing any scenario, any implementation that overrides the abstract method would work if we changed the method to virtual. We're just allowing derived classes to omit the implementation from now on. In [Types of breaking changes](https://docs.microsoft.com/en-us/dotnet/core/compatibility/#members), it also mentions that:
> ✔️ ALLOWED: Changing a member from abstract to virtual

I don't really feel strongly about it, I just think that it could make future operations a bit shorter and with less repeated code. Maybe this is a discussion for another PR.

Reply:
Ah good point - I was thinking about it the opposite way (adding an additional abstract method)